ldap/servers/slapd/connection.c

```diff
@@ -457,12 +458,20 @@ init_op_threads()
 
     work_q_stack = PR_CreateStack("connection_work_q");
     op_stack = PR_CreateStack("connection_operation");
+    alloc_per_thread_snmp_vars(max_threads);
+    init_thread_private_snmp_vars();
+    
 
+    threads_indexes = (int32_t *) slapi_ch_calloc(max_threads, sizeof(int32_t));
+    for (size_t i = 0; i < max_threads; i++) {
+        threads_indexes[i] = i + 1; /* idx 0 is reserved for global snmp_vars */
+    }
+    
     /* start the operation threads */
     for (size_t i = 0; i < max_threads; i++) {
         PR_SetConcurrency(4);
         if (PR_CreateThread(PR_USER_THREAD,
-                            (VFP)(void *)connection_threadmain, NULL,
+                            (VFP)(void *)connection_threadmain, (void *) &threads_indexes[i],
```

Comment:
Similar, off by one, you aren't reserving 0 for global since this iters from 0. You may want to make max_threads + 1 the global count instead.

Reply:
connection_threadmain (workers) are using indexes 1..<#workers> in per_thread_snmp_vars

Reply:
Also same confusion as above, all good.